stats/src/main/java/com/facebook/airlift/stats/QuantileDigest.java

```diff
@@ -1315,4 +1337,172 @@ public String toString()
 
         double middle(long lowerBound, long upperBound);
     }
+
+    public static class QuantileDigestIterator
+    {
+        private final int[] lefts; // left children
+        private final int[] rights; // right children
+        private final int[] stack; // array stack
+        private final double[] counts; // counts of nodes
+        private final byte[] levels; // levels in tree of nodes
+        private final long[] values; // values of nodes
+        private final double weightedCount; // total count of tree nodes
+        private final long min;
+        private final long max;
+        private final BitSet seen;
```

Comment:
nit: visited? also why use `BitSet` instead of `boolean[]` given the marjoiry memory is probably taken by other long/int/double arrays?

Reply:
Because there may be N iterators per qdigest, I opted for the lower allocation cost of the bitset.